src/atomate2/vasp/jobs/phonons.py

```diff
@@ -0,0 +1,409 @@
+from __future__ import annotations
+
+import copy
+import logging
+import tempfile
+from dataclasses import dataclass, field
+
+import numpy as np
+from jobflow import Flow, Response, job
+from phonopy import Phonopy
+from phonopy.phonon.band_structure import get_band_qpoints_and_path_connections
+from phonopy.structure.symmetry import elaborate_borns_and_epsilon
+from phonopy.units import VaspToTHz
+from pymatgen.core import Structure
+from pymatgen.io.phonopy import (
+    get_ph_bs_symm_line,
+    get_ph_dos,
+    get_phonopy_structure,
+    get_pmg_structure,
+)
+from pymatgen.io.vasp import Kpoints
+from pymatgen.phonon.plotter import PhononBSPlotter, PhononDosPlotter
+from pymatgen.symmetry.analyzer import SpacegroupAnalyzer
+from pymatgen.symmetry.bandstructure import HighSymmKpath
+from pymatgen.transformations.advanced_transformations import (
+    CubicSupercellTransformation,
+)
+
+from atomate2 import SETTINGS
+from atomate2.common.schemas.math import Matrix3D
+from atomate2.vasp.jobs.base import BaseVaspMaker
+from atomate2.vasp.schemas.phonons import PhononBSDOSDoc
+from atomate2.vasp.sets.base import VaspInputSetGenerator
+from atomate2.vasp.sets.core import StaticSetGenerator
+
+logger = logging.getLogger(__name__)
+
+__all__ = [
+    "generate_phonon_displacements",
+    "run_phonon_displacements",
+    "generate_frequencies_eigenvectors",
+    "PhononDisplacementMaker",
+]
+
+
+def get_phonon_object(
+    displacement, min_length, structure, sym_reduce, symprec, conventional
+):
+    transformation = CubicSupercellTransformation(min_length=min_length)
```

Comment:
Rather than having this here (where `CubicSupercellTransformation` will get called multiple times at different parts of the workflow), could it be refactored so that the supercell matrix is calculated once at the beginning of the flow and then passed directly to the various other jobs throughout.

Reply:
Yes, agreed. I will indeed change this.